actors/builtin/multisig/multisig_test.go

```diff
@@ -75,7 +76,166 @@ func TestConstruction(t *testing.T) {
+			Params:   nilParams,
+			Approved: []addr.Address{anne},
+		})
+	})
+
+	t.Run("propose with threshold met", func(t *testing.T) {
+		const numApprovals = int64(1)
+
+		rt := builder.WithBalance(abi.NewTokenAmount(20), abi.NewTokenAmount(0)).Build()
+
+		actor.verifyConstruct(rt, numApprovals, noUnlockDuration, signers...)
+
+		rt.ExpectSend(chuck, builtin.MethodSend, nilParams, sendValue, nil, 0)
+
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		actor.verifyPropose(rt, chuck, sendValue, builtin.MethodSend, nilParams)
+
+		actor.assertTransactions(rt)
+	})
+
+	t.Run("fail propose with threshold met and insufficient balance", func(t *testing.T) {
+		const numApprovals = int64(1)
+
+		rt := builder.WithBalance(abi.NewTokenAmount(0), abi.NewTokenAmount(0)).Build()
+
+		actor.verifyConstruct(rt, numApprovals, noUnlockDuration, signers...)
+
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		rt.ExpectAbort(exitcode.ErrInsufficientFunds, func() {
+			actor.verifyPropose(rt, chuck, sendValue, builtin.MethodSend, nilParams)
+		})
+
+		actor.assertTransactions(rt, multisig.MultiSigTransaction{
+			To:       chuck,
+			Value:    sendValue,
+			Method:   builtin.MethodSend,
+			Params:   nilParams,
+			Approved: []addr.Address{anne},
+		})
+	})
+
+	t.Run("fail propose from non-signer", func(t *testing.T) {
+		// non-signer address
+		richard := newIDAddr(t, 105)
+		const numApprovals = int64(2)
+
+		rt := builder.Build()
+
+		actor.verifyConstruct(rt, numApprovals, noUnlockDuration, signers...)
+
+		rt.SetCaller(richard, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		rt.ExpectAbort(exitcode.ErrForbidden, func() {
+			actor.verifyPropose(rt, chuck, sendValue, builtin.MethodSend, nilParams)
+		})
+
+		actor.assertTransactions(rt)
+	})
+}
+
+//
+// Helper methods for calling multisig actor methods
+//
+
+type msActorHarness struct {
+	multisig.MultiSigActor
+	t testing.TB
+}
+
+func (h *msActorHarness) verifyConstruct(rt *mock.Runtime, numApprovalsThresh, unlockDuration int64, signers ...addr.Address) {
+	constructParams := multisig.ConstructorParams{
+		Signers:               signers,
+		NumApprovalsThreshold: numApprovalsThresh,
+		UnlockDuration:        abi.ChainEpoch(unlockDuration),
+	}
+
+	rt.ExpectValidateCallerAddr(builtin.InitActorAddr)
+	constructRet := h.MultiSigActor.Constructor(rt, &constructParams)
+	assert.Equal(h.t, runtime.EmptyReturn{}, *constructRet)
+	rt.Verify()
+}
+
+func (h *msActorHarness) verifyPropose(rt *mock.Runtime, to addr.Address, value abi.TokenAmount, method abi.MethodNum, params abi.MethodParams) {
+	proposeParams := &multisig.ProposeParams{
+		To:     to,
+		Value:  value,
+		Method: method,
+		Params: params,
+	}
+	h.MultiSigActor.Propose(rt, proposeParams)
+	rt.Verify()
```

Comment:
This isn't verifying anything, so let's not (and rename the method to `propose`)